document domain definition documentation capabilities (#261)
btpworks/chronicle: docs/domain_modeling.md

[building chronicle](./building.md) will produce a well-typed API for your
domain. The next step is then [recording provenance](./recording_provenance.md).

## Chronicle Domain Definition Documentation Capabilities

Chronicle's documentation capabilities provide users with a powerful tool for
generating Rust code documentation and GraphQL schema documentation for their
domains.

Chronicle allows users to define a domain and generate documentation for that
domain using documentation comments in the domain.yaml file. When Chronicle
loads a valid domain.yaml file, it generates the domain on the basis of that
file, and any `doc` field comments -- they are optional fields that can be
left out -- found in the domain.yaml will be generated as Rust code
documentation as well as documentation of the GraphQL schema for the given

> Chronicle domain.

Reviewer: I'm finding this sentence a bit awkward. I'm not sure that the doc field comments are "generated" as other documentation, and the parenthetical bit about leaving optional parts out means a fair bit of parse context to reload to understand the last part of the sentence.

[building chronicle](./building.md) will produce a well-typed API for your
domain. The next step is then [recording provenance](./recording_provenance.md).

## Chronicle Domain Definition Documentation Capabilities

Chronicle's documentation capabilities provide users with a powerful tool for
generating Rust code documentation and GraphQL schema documentation for their
domains.

Chronicle provides users with the ability to define a domain by using a
domain.yaml file and adding documentation comments to it. Upon loading a
valid domain.yaml file, Chronicle generates the domain based on its contents.

> Any `doc` field comments included in the domain.yaml file are automatically
> used as documentation for both the Rust code and the GraphQL schema associated
> with the Chronicle domain. These comments are not generated but are simply
> utilized as documentation. The `doc` and `roles_doc` field comments are

Reviewer: ```suggestion
Any `doc` and `roles_doc` field comments included in the domain.yaml file are
used to generate documentation for both the Rust code and the GraphQL schema associated
with the Chronicle domain. The field comments are
```
